activerecord/lib/active_record/tasks/sqlite_database_tasks.rb

```diff
@@ -38,7 +38,16 @@ def charset
       def structure_dump(filename, extra_flags)
         dbfile = configuration["database"]
         flags = extra_flags.join(" ") if extra_flags
-        `sqlite3 #{flags} #{dbfile} .schema > #{filename}`
+
+        ignore_tables = ActiveRecord::SchemaDumper.ignore_tables
+        if ignore_tables.any?
+          tables = `sqlite3 #{dbfile} .tables`.split - ignore_tables
+          condition = tables.map { |table| "tbl_name = '#{table}'" }.join(" OR ")
+          statement = "SELECT sql FROM sqlite_master WHERE #{condition} ORDER BY tbl_name, type DESC, name"
+          `sqlite3 #{flags} #{dbfile} "#{statement}" > #{filename}`
```

Comment:
We should be anyway, but this level of control over the contents of the string makes it even more important for this to be using the array form, instead of the string-construction form, of command invocation.